src/test/java/com/example/demo/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,180 @@
+        Assertions.assertEquals(3, actual.length);
+        Assertions.assertEquals(expected, Arrays.stream(actual).toList());
+    }
+
+    @Sql(scripts = "classpath:database/books/add-categories-to-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/delete-categories-from-db.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(authorities = {"ADMIN"})
+    @Test
+    void updateCategory_validCategory_ok() throws Exception {
+        CategoryDto expected = new CategoryDto();
+        expected.setName("new updated Category");
+
+        String jsonRequest = objectMapper.writeValueAsString(expected);
+
+        MvcResult mvcResult = mockMvc.perform(put("/api/categories/1")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+
+        CategoryDto actual = objectMapper
+                .readValue(mvcResult.getResponse().getContentAsString(), CategoryDto.class);
+        Assertions.assertNotNull(actual);
+        EqualsBuilder.reflectionEquals(expected, actual);
+    }
+
+    @Sql(scripts = "classpath:database/books/add-categories-to-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/delete-categories-from-db.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(authorities = {"ADMIN"})
+    @Test
+    void updateCategory_invalidCategory_notOk() throws Exception {
+        CategoryDto expected = new CategoryDto();
+        expected.setName("new updated Category");
+
+        String jsonRequest = objectMapper.writeValueAsString(expected);
+
+        mockMvc.perform(post("/api/categories/256")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().is4xxClientError());
+    }
+
+    @Sql(scripts = "classpath:database/books/add-books-and-category-to-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/delete-books-and-category-from-db.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(authorities = {"ADMIN"})
+    @Test
+    void getBooksByCategoryId_validCategoryId_ShouldReturnAllBooks() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(get("/api/categories/1/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper
+                .readValue(mvcResult.getResponse().getContentAsString(), BookDto[].class);
+        Assertions.assertEquals(2, actual.length);
```

Comment:
```suggestion
                .readValue(mvcResult.getResponse().getContentAsString(), BookDto[].class);
                
        assertEquals(2, actual.length);
```

Reply:
Make empty line before asserts and you can write method without Assertions class, let stay it in import. It will looks better